### PR TITLE
Issue/437/magnification bias

### PR DESCRIPTION
**Pull request by marina-ricci**

### CODE REVIEW

**Comment by marina-ricci** [comment]
This should be done now.

File: clmm/theory/generic.py [hunk 1]
```diff
@@ -15,14 +16,51 @@ def compute_reduced_shear_from_convergence(shear, convergence):
...
+    alpha : array like
+            Source cummulative number density slope
+
+    Returns
+    -------
+    magnification bias : array_like
+            magnification bias
+    """
+    magnification_bias_from_magnification = np.array(
+        magnification)**(np.array([alpha]).T - 1)
```
[the remark this answers is not in the document]
Comment (marina-ricci) [reply clmm/theory/generic.py:68]: Ok, it should be done now.

File: clmm/theory/func_layer.py [hunk 2]
```diff
@@ -9,12 +9,12 @@
 import numpy as np
 
 from . import generic
-from . generic import compute_reduced_shear_from_convergence
+from . generic import *
```
Comment (m-aguena) [inline clmm/theory/func_layer.py:12]: please do not import `*`

File: clmm/theory/generic.py [hunk 3]
```diff
@@ -15,14 +17,53 @@ def compute_reduced_shear_from_convergence(shear, convergence):
 
     Parameters
     ----------
-    shear : array_like
-        Shear
-    convergence : array_like
-        Convergence
+    shear : array_like, float
+            Shear
```
Comment (m-aguena) [inline clmm/theory/generic.py:21]: there is an extra tab here

Comment (m-aguena) [reply clmm/theory/generic.py:21]: also on lines 23 and 28

File: clmm/theory/parent_class.py [hunk 4]
```diff
@@ -281,7 +283,7 @@ def eval_convergence(self, r_proj, z_cl, z_src):
             \kappa = \frac{\Sigma}{\Sigma_{crit}}
 
         or
-
+
```
Comment (m-aguena) [inline clmm/theory/parent_class.py:286]: extra tab

File: clmm/theory/parent_class.py [hunk 5]
```diff
@@ -374,3 +376,34 @@ def _eval_magnification(self, r_proj, z_cl, z_src):
         kappa = self.eval_convergence(r_proj, z_cl, z_src)
         gamma_t = self.eval_tangential_shear(r_proj, z_cl, z_src)
         return 1./((1-kappa)**2-abs(gamma_t)**2)
+    
+    def eval_magnification_bias(self, r_proj, z_cl, z_src, alpha):
```
Comment (m-aguena) [inline clmm/theory/parent_class.py:380]: can you add an validation function? just follow the model of `eval_magnification`

File: clmm/theory/parent_class.py [hunk 6]
```diff
@@ -2,7 +2,8 @@
 CLMModeling abstract class
 """
 import numpy as np
-from .generic import compute_reduced_shear_from_convergence
+import warnings
+from .generic import *
```
Comment (m-aguena) [inline clmm/theory/parent_class.py:6]: please do not import `*`

**Review by m-aguena — CHANGES_REQUESTED** [review]
Can you make sure the lines are shorter than 100 characters? You can check in the files you changed with the command `grep -x '.\{100,\}' my_file.py`

**Comment by marina-ricci** [comment]
@m-aguena , I think it should be good now!

**Review by m-aguena — APPROVED** [review]

**Review by m-aguena — APPROVED** [review]
@marina-ricci it looks good for me now.
I would just like @combet to have a final look before merging as I made a couple of changes.

**Comment by marina-ricci** [comment]
@hsinfan1996  Please be careful because the checks are failing since your commit.

**Comment by hsinfan1996** [comment]
> @hsinfan1996 Please be careful because the checks are failing since your commit.

I noticed, but I really don't think the check failures were related to my commit.

**Comment by combet** [comment]
The error appears when installing CLMM during CI (that's a new one...)

error: numpy 1.17.4 is installed but numpy>=1.18 is required by {'astropy'}
Error: Process completed with exit code 1.

**Comment by hsinfan1996** [comment]
@m-aguena astropy 5.0 is installed during `install prereq using pip`. Too new.

**Comment by hsinfan1996** [comment]
Need to edit requirements.txt.

**Review by combet — APPROVED** [review]
Thank you, all looks good!

**Pull request merged**